### PR TITLE
Replace pycrypto

### PR DESCRIPTION
**Pull request by vladmonea**
Since pycrypto is unmaintained, we're replacing the methods used in our signing and signature verification workflows.
We're also updating the code styling.

### CODE REVIEW

File: hubblestack/utils/signing.py [hunk 1]
```diff
@@ -63,39 +61,47 @@
 from cryptography.hazmat.backends import default_backend
 from cryptography.hazmat.primitives.serialization import load_pem_private_key
 
-MANIFEST_RE = re.compile(r'^\s*(?P<digest>[0-9a-fA-F]+)\s+(?P<fname>.+)$')
+MANIFEST_RE = re.compile(r"^\s*(?P<digest>[0-9a-fA-F]+)\s+(?P<fname>.+)$")
 log = logging.getLogger(__name__)
 
-def _our_byte_string(x, charmap='utf-8'):
```
Comment (grogsaxle) [inline hubblestack/utils/signing.py:69]: I'm not sure we want to drop the charmap argument here.   I understand that it is "unused", but may make for a runtime error if we do not include it.

File: hubblestack/utils/signing.py [hunk 2]
```diff
@@ -124,35 +131,36 @@ def check_verif_timestamp(target, dampener_limit=None):
...
     if ts_0 is None:
         ts_0 = time()
-        verif_log_timestamps[target] = ts_0
+        VERIFY_LOG_TIMESTAMPS[target] = ts_0
         return True
 
     ts_1 = time()
     # make a timedelta from the loaded ts vs now.
-    td =  ts_1 - ts_0
+    td = ts_1 - ts_0  # pylint: disable=invalid-name ; this is fine
```
Comment (grogsaxle) [inline hubblestack/utils/signing.py:150]: 😆

File: hubblestack/utils/signing.py [hunk 3]
```diff
@@ -420,68 +434,75 @@ def __init__(self, public_crt=None, ca_crt=None, extra_crt=None):
...
+                    message,
+                )
 
             # add to list of keys we'll use to check signatures:
             self.public_crt.append(self.PublicCertObj(i, digest, status))
 
 
 def stringify_ossl_cert(a_cert_obj):
-    """ try to stryingy a cert object into its subject components and digest hexification.
+    """try to stryingy a cert object into its subject components and digest hexification.
```
Comment (grogsaxle) [inline hubblestack/utils/signing.py:452]: Please fix spelling of stringify

File: hubblestack/utils/signing.py (hunk 1 above)
Comment (grogsaxle) [reply hubblestack/utils/signing.py:69]: Maybe add this: 
# pylint: disable=unused-argument

File: hubblestack/utils/signing.py [hunk 4]
```diff
@@ -111,8 +117,9 @@ def check_is_ca(crt):
         pass
     return False
 
+
 def check_verif_timestamp(target, dampener_limit=None):
```
Comment (grogsaxle) [inline hubblestack/utils/signing.py:121]: Please rename to 'check_verify_timestamp'

File: hubblestack/utils/signing.py [hunk 5]
```diff
@@ -741,24 +844,24 @@ def verify_files(targets, mfname=None, sfname=None,
...
                 matched = MANIFEST_RE.match(line)
                 if matched:
-                    digest,manifested_fname = matched.groups()
+                    digest, manifested_fname = matched.groups()
                     manifested_fname = normalize_path(manifested_fname)
                     if manifested_fname in digests:
                         digests[manifested_fname] = digest
     # number of seconds before a FAIL or UNKNOWN is set to the returner
-    global verif_log_timestamps
+    global VERIFY_LOG_TIMESTAMPS
```
Comment (grogsaxle) [inline hubblestack/utils/signing.py:864]: why `global` here?

File: hubblestack/utils/signing.py [hunk 6]
```diff
@@ -508,16 +529,56 @@ def manifest(targets, mfname=None):
...
+    :returns: PEM encoded string
+    """
+    output = "-----BEGIN {}-----\n".format(marker)
+    chunks = [b2a_base64(data[i : i + 48]).decode("latin-1") for i in range(0, len(data), 48)]
+    output += "".join(chunks)
+    output += "-----END {}-----".format(marker)
+    return output
+
+
+def decode_pem(data: bytes) -> str:
```
Comment (grogsaxle) [inline hubblestack/utils/signing.py:557]: May be able to use this instead: https://cryptography.io/en/latest/x509/reference/#loading-certificates

File: hubblestack/utils/signing.py (hunk 5 above)
Comment (vladmonea) [reply hubblestack/utils/signing.py:864]: i'll try to kill it. with fire

File: hubblestack/utils/signing.py (hunk 2 above)
Comment (vladmonea) [reply hubblestack/utils/signing.py:150]: can we keep him?

File: hubblestack/utils/signing.py (hunk 6 above)
Comment (vladmonea) [reply hubblestack/utils/signing.py:557]: I'm not sure. `PEM.encode` and `PEM.decode` were being used to get a `base64` representation of the key/cert. The functions from `cryptography.x509` are highly specialised in what they do.

File: hubblestack/utils/signing.py [hunk 7]
```diff
@@ -272,27 +283,28 @@ class X509AwareCertBucket:
...
+            if hasattr(HPPC, "public_crt"):
+                log.debug("using pre-packaged-public_crt")
                 public_crt = HPPC.public_crt
-            if hasattr(HPPC, 'ca_crt'):
-                log.debug('using pre-packaged-ca_crt')
+            if hasattr(HPPC, "ca_crt"):
+                log.debug("using pre-packaged-ca_crt")
                 ca_crt = HPPC.ca_crt
         except ImportError:
             pass
```
Comment (mzainea) [inline hubblestack/utils/signing.py:310]: should this really pass silently?

Comment (vladmonea) [reply hubblestack/utils/signing.py:310]: i'm not sure this should be fixed in a PR addressing a totally different thing. otherwise, fair point

Comment (vladmonea) [reply hubblestack/utils/signing.py:310]: @grogsaxle should we fix this in a different PR, or since we're here might as well?

**Comment by vladmonea** [comment]
Closing PR because untangling the circular dependencies is too much hassle to be worth the effort

**Pull request closed without merging**